### PR TITLE
Apache Arrow recipe

### PR DESCRIPTION
**Pull request by maccam912**
Had to create the PR manually. Am I doing it right?

### CODE REVIEW

**Comment by giordano** [comment]
> Am I doing it right?

Sure, no problem opening a pull request manually! However note that there is already #918 building the same package

**Comment by maccam912** [comment]
Cool. The other one seems to have stalled a bit, I'll see if I can get the other checks working.

File: A/Arrow/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+make
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline A/Arrow/build_tarballs.jl:26]: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=5588&view=logs&j=01859c06-840e-59ce-0724-571b2b074689&t=fc66dd4d-34e4-5ce9-a2f5-450f663f0150&l=1035

```
┌ Warning: /agent/_work/2/s/A/Arrow/build/x86_64-linux-gnu/tSLp0J6a/destdir/lib/libarrow.so.100.1.0 contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/80UOF/src/auditor/compiler_abi.jl:222
```

```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: A/Arrow/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd arrow/cpp/
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+make
```
Comment (giordano) [inline A/Arrow/build_tarballs.jl:20]:
```suggestion
make -j${nproc}
```

File: A/Arrow/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+products = [
+    LibraryProduct("libarrow", :libarrow)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```
Comment (giordano) [inline A/Arrow/build_tarballs.jl:39]: Why such a recent version of GCC?  That's unlikely to work with the official binaries of Julia

Comment (giordano) [reply A/Arrow/build_tarballs.jl:39]: That's actually exactly the reason why the library can't be `dlopen`ed: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=5588&view=logs&j=01859c06-840e-59ce-0724-571b2b074689&t=fc66dd4d-34e4-5ce9-a2f5-450f663f0150&l=1046
```
/usr/local/bin/../lib/julia/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /agent/_work/2/s/A/Arrow/build/x86_64-linux-gnu/tSLp0J6a/destdir/lib/libarrow.so.100.1.0)
```
#918 uses the default GCC version, v4.8.  I don't see why we should use GCC v9 here

**Comment by maccam912** [comment]
Well the PR ended up messy as heck now, but the platforms it builds on work. Ultimately I want to get it working on windows, which it doesn't now. Should I wait to merge until I have that working, or merge the linux stuff now and come back with improvements once I have some?

**Comment by maccam912** [comment]
Nope. I closed it. I'm testing it out and I'm not happy with how it's set up, I don't want to give people false hope that this will make it easy to use Arrow.

**Pull request closed without merging**